File: app/src/main/java/com/mai/nix/maiapp/MaiApp.java
package com.mai.nix.maiapp;

import android.app.Application;
import android.arch.persistence.room.Room;

import com.mai.nix.maiapp.room.AppDatabase;

public class MaiApp extends Application {
    private static MaiApp sInstance;

    private AppDatabase mAppDatabase;

    @Override
    public void onCreate() {
        super.onCreate();
        sInstance = this;
        mAppDatabase = Room.databaseBuilder(this, AppDatabase.class, "mai_database").build();
    }

    public static MaiApp getInstance() {
        return sInstance;
    }

    public AppDatabase getDatabase() {
        return mAppDatabase;
    }
}

File: app/src/main/java/com/mai/nix/maiapp/MaiApp.kt
package com.mai.nix.maiapp

import android.app.Application
import androidx.appcompat.app.AppCompatDelegate
import androidx.room.Room
import com.mai.nix.maiapp.database.AppDatabase
import com.mai.nix.maiapp.helpers.UserSettings

class MaiApp: Application() {

    private lateinit var database: AppDatabase

    override fun onCreate() {
        super.onCreate()
        UserSettings.initialize(this)
        AppCompatDelegate.setDefaultNightMode(UserSettings.getTheme(this))
        database = Room.databaseBuilder(this, AppDatabase::class.java, "app_database").build()
    }

    fun getDatabase(): AppDatabase {
        return database
    }
}